providers/gce/gce_loadbalancer.go

```diff
@@ -134,6 +145,28 @@ func (g *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, svc
 		return nil, err
 	}
 
+	// Services with multiples protocols are not supported by this controller, warn the users and sets
+	// the corresponding Service Status Condition.
+	// https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1435-mixed-protocol-lb
+	if err := checkMixedProtocol(svc.Spec.Ports); err != nil {
+		if hasLoadBalancerPortsError(svc) {
+			return nil, err
+		}
+		klog.Warningf("Ignoring service %s/%s using different ports protocols", svc.Namespace, svc.Name)
+		g.eventRecorder.Event(svc, v1.EventTypeWarning, LoadBalancerPortsErrorReason, "LoadBalancers with multiple protocols are not supported.")
+		svcApplyStatus := corev1apply.ServiceStatus().WithConditions(
+			metav1apply.Condition().
+				WithType(v1.LoadBalancerPortsError).
+				WithStatus(metav1.ConditionTrue).
+				WithReason(LoadBalancerPortsErrorReason).
+				WithMessage("LoadBalancer with multiple protocols are not supported"))
+		svcApply := corev1apply.Service(svc.Name, svc.Namespace).WithStatus(svcApplyStatus)
+		if _, errApply := g.client.CoreV1().Services(svc.Namespace).ApplyStatus(ctx, svcApply, metav1.ApplyOptions{FieldManager: "gce-legacy-cloud-controller", Force: true}); errApply != nil {
+			return nil, errApply
+		}
+		return nil, err
```

Comment:
that's a little repeated code between create and update

why not define private method
```
func (g *Cloud) applyLoadBalancerPortsErrorStatus(ctx context.Context, svc *v1.Service) error {
	klog.Warningf("Ignoring service %s/%s using different ports protocols", svc.Namespace, svc.Name)
	g.eventRecorder.Event(svc, v1.EventTypeWarning, LoadBalancerPortsErrorReason, "LoadBalancers with multiple protocols are not supported.")
	svcApplyStatus := corev1apply.ServiceStatus().WithConditions(
		metav1apply.Condition().
			WithType(v1.LoadBalancerPortsError).
			WithStatus(metav1.ConditionTrue).
			WithReason(LoadBalancerPortsErrorReason).
			WithMessage("LoadBalancer with multiple protocols are not supported"))
	svcApply := corev1apply.Service(svc.Name, svc.Namespace).WithStatus(svcApplyStatus)
	if _, errApply := g.client.CoreV1().Services(svc.Namespace).ApplyStatus(ctx, svcApply, metav1.ApplyOptions{FieldManager: "gce-legacy-cloud-controller", Force: true}); errApply != nil {
		return errApply
	}
	return nil
}
```

Reply:
it is not worth it to encapsulate it mainly when the goal has to remove it, I just add it to avoid users having a broke load balancer without noticing it

Reply:
also I do not want to drift from k/k